server/src/main/java/org/elasticsearch/index/fieldvisitor/FieldsVisitor.java

```diff
@@ -87,9 +87,9 @@ public Status needsField(FieldInfo fieldInfo) {
                 : Status.NO;
     }
 
-    public void postProcess(MapperService mapperService) {
+    public final void postProcess(Function<String, MappedFieldType> fieldTypeLookup) {
```

Comment:
I do wonder if we can remove this step entirely, or replace it with the ValueFetcher functionality somehow - for a followup though.

Reply:
So this doesn't do anything to the `_id`, `_source` or `_routing` fields, so I think we can remove the call to it entirely from `LuceneChangesSnapshot` and possibly a few other places as well.

Reply:
ok I will look at it, possibly make this change in a separate PR then.

Reply:
++